lib/_http_outgoing.js

```diff
@@ -90,7 +90,7 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
+  if (typeof callback === 'function') {
```

Comment:
although ```if (typeof callback === 'function')``` will still evaluate false whether or not callback is defined.

Seem like the extra if statement here isn't neccessary